### PR TITLE
feat(search): add banners and promoteds

### PR DESCRIPTION
**Pull request by tiborux**
EX-4299

### CODE REVIEW

**Review by tajespasarela — CHANGES_REQUESTED** [review]
hide the banner and promoted title at the moment, meanwhile we don't have designs:

in `App.vue`

```
  .x-banner,
  .x-promoted {
    &__title {
      display: none;
    }
  }
```

File: src/components/main.vue [hunk 1]
```diff
@@ -80,20 +80,24 @@
...
+              </template>
+              <template #Banner="{ item: banner }">
+                <Banner :banner="banner" />
+              </template>
 
+              <template #Promoted="{ item: promoted }">
+                <Promoted :promoted="promoted" />
+              </template>
+            </BaseVariableColumnGrid>
+          </PromotedLists>
```
Comment (LuisMartinez15) [inline src/components/main.vue:98]: Change to `PromotedsList`

```suggestion
          <PromotedsList>
            <BaseVariableColumnGrid class="x-grid" :animation="resultsAnimation">
              <template #Result="{ item: result }">
                <Result :result="result" />
              </template>
              <template #Banner="{ item: banner }">
                <Banner :banner="banner" />
              </template>
              <template #Promoted="{ item: promoted }">
                <Promoted :promoted="promoted" />
              </template>
            </BaseVariableColumnGrid>
          </PromotedsList>
```

File: src/adapter/adapter.ts [hunk 2]
```diff
@@ -73,5 +79,8 @@ export const adapter = new EmpathyAdapterBuilder()
   .onBeforeRequest(({ request }) => {
     delete request.scope;
   }, 'topRecommendations')
+  .onBeforeRequest(({ request }) => {
+    delete request.scope;
+  }, 'search')
```
Comment (LuisMartinez15) [inline src/adapter/adapter.ts:84]: Do we truly want this? Aren't banners and promoted associated with a one or multiple scopes?

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

File: src/adapter/adapter.ts (hunk 2 above)
Comment (tiborux) [reply src/adapter/adapter.ts:84]: the request is not returning banners or promoted if you send the scope

Comment (LuisMartinez15) [reply src/adapter/adapter.ts:84]: As discussed with @tajespasarela, this parameter should not be removed, as banners and promoted should be defined per one or multiple scopes.

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

**Review by LuisMartinez15 — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**